.circleci/config.yml

```diff
@@ -149,25 +146,29 @@ jobs:
     steps:
       - checkout
       - restore_module_cache
-      - persist_to_workspace:
-          root: ~/
-          paths: project
+      - run:
+          name: Install deps
+          command: make -j8 for-all-make CMD='dep'
       - run:
           name: Install tools
           command: make install-tools
       - run:
           name: Install testbed tools
           command: make -C testbed install-tools
+      - save_module_cache
       - persist_to_workspace:
           root: ~/
-          paths: go/bin
+          paths:
+            - project
+            - go/bin
+            - go/pkg/mod
   lint:
     executor: golang
     steps:
       - attach_to_workspace
       - run:
           name: Lint
-          command: CMD="make lint" make -j4 for-all
+          command: CMD="lint" make -j4 for-all-make
```

Comment:
nit: similar to how you did deps above I think it reads better for CMD="foo" to come at the end.